src/freenas/usr/local/lib/middlewared_truenas/plugins/failover_/event.py

```diff
@@ -519,6 +519,14 @@ def carp_master(self, fobj, ifname, vhid, event, user_override, forcetakeover):
                 if attach_all_job.error:
                     logger.error('Failed to attach geli providers: %s', attach_all_job.error)
 
+                try:
+                    self.middleware.call_sync('disk.sed_unlock_all')
```

Comment:
I wonder if there may be any implications here.

In theory this only needs to be called once, on boot. There is really no reason to call that when actually taking over, since disks will already be unlocked.

Reply:
We can't call this at boot time because we don't know which controller is `MASTER` or `BACKUP`. That's why I'm calling it in the event script.

Reply:
You must have understood me. Yes, I got that, but I mean we will call it every time a takeover happens, not only on boot.

Reply:
I don't see a way around that. `sed_unlock_all` will call `sedutil --query` to see if the disk(s) is locked before trying to unlock it? There will be a significant delay in failover event if this is called and the disks end up being locked. (i.e. both controllers are powered on and HA is configured). But again, I don't see a way around that?

Reply:
The way around that would be somehow have a flag around to only call it if we are in the boot process, prior to enabling CARP.